maintenance/projects/js-toolkit/packages/liferay-npm-build-tools-common/src/namespace.ts

```diff
@@ -109,5 +109,9 @@ export function getNamespace(moduleName: string): string {
  * @return the namespace for modules
  */
 export function makeNamespace({name}: {name: string}): string {
+	if (name.startsWith('@')) {
```

Comment:
This converts `@liferay/frontend-js-web` into `liferay!frontend-js-web`.

Even though it looks hacky, it's needed since our bundler namespaces should not contain any `/` to avoid misleading the web server and all the code we wrote since before this thing even existed.

For the very same reason we had to introduce the `$` notation when bundler 2 namespaces appeared. But this time we need to use a different char.

Currently, there is a subset of characters that can be used in URLs and filesystems (windows and unix) but it's illegal in npm packages. Among those are ! and $, which is the reason why we choose them. 

If we chose legal npm characters we would have the possibility to have collisions. For example, imagine we chose `_` instead of `!`, then `@liferay/frontend` would be translated to  `liferay_frontend` which would collide with the `liferay_frontend` package.